### PR TITLE
Added flail test

### PR DESCRIPTION
**Pull request by PraveenPenguin**
added flail test which does system call fuzz

Signed-off-by: Praveen K Pandey <praveen@linux.vnet.ibm.com>

### CODE REVIEW

File: fs/flail.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    def setUp(self):
+        '''
+        Setup Flail
+        '''
+        smm = SoftwareManager()
+        for package in ['gcc', 'make']:
+            if not smm.check_installed(package) and not smm.install(package):
+                self.cancel(package + ' is needed for the test to be run')
+
+        tarball = os.path.join(self.datadir, "flail-0.2.0.tar.gz")
```
Comment (narasimhan-v) [inline fs/flail.py]: Is flail not maintained in upstream ?

File: fs/flail.py [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+    def test(self):
+        '''
+        Runs flail with the appropriate parameters.
+
+        :param fstype: Filesystem type there user want to run flail
+        '''
+        self.clear_dmesg()
+        os.chdir(self.build_dir)
+        fs_type = self.params.get('fstype', default='xfs')
```
Comment (narasimhan-v) [inline fs/flail.py]: Better to read yaml in setUp()

File: fs/flail.py [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+        process.system('./flail %s 1' % fs_type)
+        dmesg = process.system_output('dmesg')
+        match = re.search(r'Call Trace:', dmesg, re.M | re.I)
+        if match:
+            self.log.info("some call traces seen please check")
```
Comment (narasimhan-v) [inline fs/flail.py]: I think we can FAIL, or at least WARN, in such cases. Your thoughts ?

File: fs/flail.py (hunk 1 above)
Comment (PraveenPenguin) [reply fs/flail.py]: @narasimhan-v no we got this test from internal

File: fs/flail.py (hunk 2 above)
Comment (PraveenPenguin) [reply fs/flail.py]: sure will do

File: fs/flail.py (hunk 3 above)
Comment (PraveenPenguin) [reply fs/flail.py]: @narasimhan-v sure i miss that will change to `self.fail` thanks for catching this

File: fs/flail.py (hunk 1 above)
Comment (narasimhan-v) [reply fs/flail.py]: @PraveenPenguin my suggestion would be to have it in upstream (maybe create a git if owner is fine). If that does not work out, then we can have tarball here.

File: fs/flail.py [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+    def test(self):
+        '''
+        Runs flail with the appropriate parameters.
+
+        :param fstype: Filesystem type there user want to run flail
+        '''
+        self.clear_dmesg()
+        os.chdir(self.build_dir)
+        fs_type = self.params.get('fstype', default='xfs')
+        process.system('./flail %s 1' % fs_type)
```
Comment (harish-24) [inline fs/flail.py]: What does this second input `1` signify to? If it is significant, better to move to yaml.

Comment (harish-24) [inline fs/flail.py]: And if the command fails, test would error. Better to catch it or use `ignore_status` if we can skip it.

File: fs/flail.py (hunk 1 above)
Comment (narasimhan-v) [reply fs/flail.py]: @PraveenPenguin We can contact the maintainer, and if no response, then we can push this commit.
How long can we wait for the maintainer's response ? any suggestions ?

Comment (PraveenPenguin) [reply fs/flail.py]: @narasimhan-v i think we can have it here only as i did not get any response from maintainer , will send V2 as soon as possible

**Comment by PraveenPenguin** [comment]
@narasimhan-v done with changes can you please have a look

File: fs/flail.py (hunk 4 above)
Comment (narasimhan-v) [reply fs/flail.py]: @PraveenPenguin any conclusion here ?

**Comment by narasimhan-v** [comment]
@PraveenPenguin Thanks for the patch, @harish-24 Thanks for the review.

**Pull request merged**